sdk/core/Azure.Core/src/ResourceIdentifier.cs

```diff
@@ -385,7 +374,65 @@ public override int GetHashCode()
         /// <returns> True if the left object is greater than or equal to the right. </returns>
         public static bool operator >=(ResourceIdentifier left, ResourceIdentifier right)
         {
-            return ReferenceEquals(left, null) ? ReferenceEquals(right, null) : left.CompareTo(right) >= 0;
+            return left is null ? right is null : left.CompareTo(right) >= 0;
+        }
+
+        private static List<string> Trim(List<string> list, int numberToTrim)
+        {
+            if (numberToTrim < 0 || numberToTrim > list.Count)
+                throw new ArgumentOutOfRangeException(nameof(numberToTrim));
+            list.RemoveRange(0, numberToTrim);
+            return list;
+        }
```

Comment:
It doesn't look like we are adding anything here but the default start index (RemoveRange already does the same validation and will throw). Should we just call RemoveRange directly?